res/controllers/Pioneer-DDJ-400-script.js

```diff
@@ -0,0 +1,741 @@
+PioneerDDJ400.loopAdjustMultiply = 50;
+
+// Beatjump pad (beatjump_size values)
+PioneerDDJ400.beatjumpPad = {
+    0x20: -1, // PAD 1
+    0x21: 1,  // PAD 2
+    0x22: -2, // PAD 3
+    0x23: 2,  // PAD 4
+    0x24: -4, // PAD 5
+    0x25: 4,  // PAD 6
+    0x26: -8, // PAD 7
+    0x27: 8   // PAD 8
+};
+
+// For controls that need it (e.g., tempo sliders)
+PioneerDDJ400.highResMSB = {
+    "[Channel1]": {},
+    "[Channel2]": {}
+};
+
+PioneerDDJ400.trackLoadedLED = function(value, group, _control) {
+    if (value) {
+        value = 0x7F;
+    } else {
+        value = 0x00;
+    }
+    var channel = group.match(/^\[Channel(\d+)\]$/)[1];
+    midi.sendShortMsg(0x9F, 0x00+(channel-1), value);
+};
+
+PioneerDDJ400.toggleLight = function(midiIn, active) {
+    midi.sendShortMsg(midiIn.status, midiIn.data1, active ? 0x7F : 0);
+};
+
+//
+// Init
+//
+
+PioneerDDJ400.init = function() {
+    // init controller
+
+    // show focus buttons on Effect Ract 1 only
+    engine.setValue("[EffectRack1_EffectUnit1]", "show_focus", 1);
+
+    // Connect the VU-Meter LEDS
+    engine.makeConnection("[Channel1]", "VuMeter", PioneerDDJ400.vuMeterUpdate);
+    engine.makeConnection("[Channel2]", "VuMeter", PioneerDDJ400.vuMeterUpdate);
+
+    // reset vumeter
+    PioneerDDJ400.toggleLight(PioneerDDJ400.lights.deck1.vuMeter, false);
+    PioneerDDJ400.toggleLight(PioneerDDJ400.lights.deck2.vuMeter, false);
+
+    // enable soft takeover for rate controls and FX level/depth
+    engine.softTakeover("[Channel1]", "rate", true);
+    engine.softTakeover("[Channel2]", "rate", true);
+    engine.softTakeover("[EffectRack1_EffectUnit1_Effect1]", "meta", true);
+    engine.softTakeover("[EffectRack1_EffectUnit1_Effect2]", "meta", true);
+    engine.softTakeover("[EffectRack1_EffectUnit1_Effect3]", "meta", true);
+    engine.softTakeover("[EffectRack1_EffectUnit1]", "mix", true);
+
+    // Sampler callbacks
+    for (var i = 1; i <= 16; ++i) {
+        engine.makeConnection("[Sampler" + i + "]", "play", PioneerDDJ400.samplerPlayOutputCallbackFunction);
+    }
+
+    // trigger "track loaded" animations when a track is loaded
+    engine.makeConnection("[Channel1]", "track_loaded", PioneerDDJ400.trackLoadedLED);
+    engine.makeConnection("[Channel2]", "track_loaded", PioneerDDJ400.trackLoadedLED);
+
+    // eye candy : play the "track loaded" animation on both decks at startup
+    midi.sendShortMsg(0x9F, 0x00, 0x7F);
+    midi.sendShortMsg(0x9F, 0x01, 0x7F);
+
+    // resets pad mode to hotcue
+    PioneerDDJ400.performancePads.reset();
+
+    // turn on loop in and out lights
+    PioneerDDJ400.setLoopButtonLights(0x90, 0x7F);
+    PioneerDDJ400.setLoopButtonLights(0x91, 0x7F);
+
+    // handle loop toggle events
+    engine.makeConnection("[Channel1]", "loop_enabled", PioneerDDJ400.loopToggle);
+    engine.makeConnection("[Channel2]", "loop_enabled", PioneerDDJ400.loopToggle);
+
+    // poll the controller for current control positions on startup
```

Comment:
```suggestion
    // query the controller for current control positions on startup
```
"poll" implies an ongoing recurring loop